File: Desktop/Services/ApiService.cs
using System;
using System.Net.Http;
using System.Net.Http.Json;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using API.Models;

namespace Desktop.Services;

public static class ApiService
{
    public static HttpClient HttpClient = new();
    public static string BaseUrl = "http://localhost:5299/api/";
    public static string Token = "";

    public static async Task<ApiResponse<T?>> Request<T>(HttpMethod method, string url, object? data = null)
    {
        try
        {
            var request = new HttpRequestMessage(method, BaseUrl + url);

            if (Token != "") request.Headers.Add("Authorization", $"Bearer {Token}");
            if (data != null) request.Content = JsonContent.Create(data);

            var response = await HttpClient.SendAsync(request);

            return await response.Content.ReadFromJsonAsync<ApiResponse<T>>();
        }
        catch (Exception ex)
        {
            Console.WriteLine($"API Error: {ex.Message}");
            return new ApiResponse<T?>(default, "Ошибка запроса", 500);
        }
    }
}

File: Web/src/services/ApiService.js
import SecureLS from "secure-ls";

export class ApiService {
    static ls = new SecureLS();

    static async request(method, url, data) {
        try {
            const response = await fetch(`http://localhost:5299/api${url}`, {
                method: method,
                headers: {
                    'Content-Type': 'application/json',
                    'Authorization': `Bearer ${this.ls.get("token")}`
                },
                body: data ? JSON.stringify(data) : undefined,
            });

            return await response.json();
        }
        catch (error) {
            console.error(error);
            return { IsSuccess: false, Error: error.message, Code: 500 };
        }
    }

    static get user() {
        return this.ls.get("user");
    }
}
